core/src/main/java/org/kiwix/kiwixmobile/core/settings/CorePrefsFragment.kt

```diff
@@ -0,0 +1,286 @@
+import org.kiwix.kiwixmobile.core.CoreApp.Companion.coreComponent
+import org.kiwix.kiwixmobile.core.CoreApp.Companion.instance
+import org.kiwix.kiwixmobile.core.NightModeConfig
+import org.kiwix.kiwixmobile.core.R
+import org.kiwix.kiwixmobile.core.main.AddNoteDialog
+import org.kiwix.kiwixmobile.core.main.CoreMainActivity
+import org.kiwix.kiwixmobile.core.utils.LanguageUtils
+import org.kiwix.kiwixmobile.core.utils.LanguageUtils.Companion.handleLocaleChange
+import org.kiwix.kiwixmobile.core.utils.SharedPreferenceUtil
+import org.kiwix.kiwixmobile.core.utils.dialog.DialogShower
+import org.kiwix.kiwixmobile.core.utils.dialog.KiwixDialog
+import org.kiwix.kiwixmobile.core.utils.dialog.KiwixDialog.OpenCredits
+import java.io.File
+import java.util.Locale
+import javax.inject.Inject
+
+abstract class CorePrefsFragment : PreferenceFragmentCompat(), SettingsContract.View,
+  OnSharedPreferenceChangeListener {
+  @JvmField @Inject
+  var presenter: SettingsPresenter? = null
+
+  @JvmField @Inject
+  var sharedPreferenceUtil: SharedPreferenceUtil? = null
+
+  @JvmField @Inject
+  var storageCalculator: StorageCalculator? = null
+
+  @JvmField @Inject
+  var nightModeConfig: NightModeConfig? = null
+
+  @JvmField @Inject
+  var alertDialogShower: DialogShower? = null
+  override fun onCreatePreferences(savedInstanceState: Bundle, rootKey: String) {
+    coreComponent
+      .activityComponentBuilder()
+      .activity(requireActivity())
+      .build()
+      .inject(this)
+    addPreferencesFromResource(R.xml.preferences)
+    setStorage()
+    setUpSettings()
+    setupZoom()
+    LanguageUtils(requireActivity()).changeFont(
+      requireActivity().layoutInflater,
+      sharedPreferenceUtil!!
+    )
+  }
+
+  private fun setupZoom() {
+    val textZoom = findPreference<Preference>(INTERNAL_TEXT_ZOOM)
+    textZoom!!.onPreferenceChangeListener =
+      Preference.OnPreferenceChangeListener { preference: Preference?, newValue: Any ->
+        sharedPreferenceUtil!!.textZoom = (newValue as Int + ZOOM_OFFSET) * ZOOM_SCALE
+        updateTextZoomSummary(textZoom)
+        true
+      }
+    updateTextZoomSummary(textZoom)
+  }
+
+  private fun updateTextZoomSummary(textZoom: Preference?) {
+    textZoom!!.summary = getString(R.string.percentage, sharedPreferenceUtil!!.textZoom)
+  }
+
+  protected abstract fun setStorage()
+  override fun onResume() {
+    super.onResume()
+    preferenceScreen.sharedPreferences
+      .registerOnSharedPreferenceChangeListener(this)
+  }
+
+  override fun onPause() {
+    super.onPause()
+    preferenceScreen.sharedPreferences
+      .unregisterOnSharedPreferenceChangeListener(this)
+  }
+
+  fun setUpSettings() {
+    setAppVersionNumber()
+  }
+
+  protected fun setUpLanguageChooser(preferenceId: String?) {
+    val languagePref = findPreference<ListPreference>(
+      preferenceId!!
+    )
+    val languageCodeList = LanguageUtils(requireActivity()).keys
+    languageCodeList.add(0, Locale.ROOT.language)
```

Comment:
Here `languageCodeList` has a data type list of strings. `List` doesn't have `add()`. Use any other data types that support this.